src/emqttd_access_control.erl

```diff
@@ -123,89 +102,71 @@ register_mod(Type, Mod, Opts) when Type =:= auth; Type =:= acl->
         [{_, Mods}] -> Mods
     end.
 
 tab_key(auth) -> auth_modules;
 tab_key(acl)  -> acl_modules.
 
-%%------------------------------------------------------------------------------
-%% @doc Stop access control server
-%% @end
-%%------------------------------------------------------------------------------
-stop() ->
-    gen_server:call(?MODULE, stop).
+%% @doc Stop access control server.
+stop() -> gen_server:call(?MODULE, stop).
 
-%%%=============================================================================
-%%% gen_server callbacks
-%%%=============================================================================
+%%--------------------------------------------------------------------
+%% gen_server callbacks
+%%--------------------------------------------------------------------
 
 init([Opts]) ->
     ets:new(?ACCESS_CONTROL_TAB, [set, named_table, protected, {read_concurrency, true}]),
-
     ets:insert(?ACCESS_CONTROL_TAB, {auth_modules, init_mods(auth, proplists:get_value(auth, Opts))}),
     ets:insert(?ACCESS_CONTROL_TAB, {acl_modules, init_mods(acl, proplists:get_value(acl, Opts))}),
-    {ok, state}.
+    {ok, #state{}}.
 
 init_mods(auth, AuthMods) ->
-    [init_mod(fun authmod/1, Name, Opts) || {Name, Opts} <- AuthMods];
+    [init_mod(authmod(Name), Opts) || {Name, Opts} <- AuthMods];
 
 init_mods(acl, AclMods) ->
-    [init_mod(fun aclmod/1, Name, Opts) || {Name, Opts} <- AclMods].
+    [init_mod(aclmod(Name), Opts) || {Name, Opts} <- AclMods].
 
-init_mod(Fun, Name, Opts) ->
-    Module = Fun(Name),
-    {ok, State} = Module:init(Opts),
-    {Module, State, 0}.
+init_mod(Mod, Opts) ->
+    {ok, State} = Mod:init(Opts), {Mod, State, 0}.
 
 handle_call({register_mod, Type, Mod, Opts, Seq}, _From, State) ->
     Mods = lookup_mods(Type),
-    Reply =
-    case lists:keyfind(Mod, 1, Mods) of
-        false ->
-            case catch Mod:init(Opts) of
-                {ok, ModState} ->
-                    NewMods =
-                    lists:sort(fun({_, _, Seq1}, {_, _, Seq2}) ->
-                                   Seq1 >= Seq2
-                               end, [{Mod, ModState, Seq} | Mods]),
-                    ets:insert(?ACCESS_CONTROL_TAB, {tab_key(Type), NewMods}),
-                    ok;
-                {'EXIT', Error} ->
-                    lager:error("Access Control: register ~s error - ~p", [Mod, Error]),
-                    {error, Error}
-            end;
-        _ ->
-            {error, existed}
-    end,
-    {reply, Reply, State};
+    Existed = lists:keyfind(Mod, 1, Mods),
+    {reply, if_existed(Existed, fun() ->
+                case catch Mod:init(Opts) of
+                    {ok, ModState} ->
+                        NewMods = lists:sort(fun({_, _, Seq1}, {_, _, Seq2}) ->
+                                            Seq1 >= Seq2
+                                    end, [{Mod, ModState, Seq} | Mods]),
+                        ets:insert(?ACCESS_CONTROL_TAB, {tab_key(Type), NewMods}),
```

Comment:
Line 152 is too long:                         ets:insert(?ACCESS_CONTROL_TAB, {tab_key(Type), NewMods}),.